openstack/keystone/templates/etc/_keystone.conf.tpl

```diff
@@ -29,7 +29,7 @@ default_tag = vc-{{ $az }}-0
 
 {{- if .Values.api.auth }}
 [auth]
-methods = {{ .Values.api.auth.methods | default "password,token,application_credential" }}
+methods = {{ if .Values.federation.oidc.enabled }}openid,{{ end }}{{ .Values.api.auth.methods | default "password,token,application_credential" }}
```

Comment:
This is very implicit IMHO, and might confuse a person reading `.Values.api.auth.methods` value and not realizing why the provided values are not in effect. I would suggest adding a comment next to the value declaration/definition, or otherwise make it explicit.

Reply:
The docs suggests to make sure that `external` is removed

[1] https://docs.openstack.org/keystone/latest/admin/external-authentication.html#configuration
> You should disable the external auth method if you are currently using federation. External auth and federation both use the REMOTE_USER variable. Since both the mapped and external plugin are being invoked to validate attributes in the request environment, it can cause conflicts.

[2] https://docs.openstack.org/keystone/latest/admin/federation/configure_federation.html#add-the-auth-method
> You should also remove external as an allowable method.

Explicit configuration and comment would best here, IMHO

Reply:
> The docs suggests to make sure that external is removed

And we cannot do that. We have to go against the official documentation unfortunately :)